finalfrontier/src/train_model.rs

```diff
@@ -101,8 +101,11 @@ impl<T> TrainModel<T> {
     }
 
     /// Get the mean input embedding of the given indices.
-    pub(crate) fn mean_input_embedding(&self, indices: &[u64]) -> Array1<f32> {
-        Self::mean_embedding(self.input.view(), indices)
+    pub(crate) fn mean_input_embedding(&self, idx: &Idx) -> Array1<f32> {
+        match idx {
+            Idx::Single(idx) => self.input.subview(Axis(0), *idx).to_owned(),
```

Comment:
`.row`

Reply:
We should then fix the other calls in `train_model`, too. The line was just copy pasted iirc.